### ISSUE TITLE
Deleting Account?

### ISSUE THREAD

**Issue by kathy-phet** [phetsims/website#252]
I was just working on testing registration, and realized that I think we offer no way to remove your own account. I know this has complexities around it ... like how do we manage submitted activities that are no longer associated with an account. 

But if it is not already on the to-be-discussed list, we should add it as it seems like we should provide users with ways to remove their accounts.

**Comment by patricialoeblein** [comment]
This is an interesting discussion. One thing we did with the recent emails to Gold star authors (about 75 people)  was find out if the contact email actually still worked. I was happy to see that few  bounced back. I imagine that the larger population of contributing teachers will have deadend contact information.  I have been glad that the activities are still available. In the cases where there is now an html5 sim, I plan to have the lessons adapted by an active contributor like the summer teachers or me or ..? So I hope that the activities would stay unless the person deletes their activities prior to canceling their account (deleting activities using the "My Activities" path). 
People may have a variety of reasons including no longer in the teaching field, but they don't mind if their lessons are still up. An example would be Amy Jordan.  If teachers post an activity and then remove their account, what would the contact field display if we don't remove the activity? Presently, an email is displayed even if it is not valid. Karen King had an invalid email and made a new account which did not change the email address on her activities. I went to each activity and corrected the contact information.

**Comment by aaronsamuel137** [comment]
Discussed on 8/6/15. We will add the ability to delete an account with the following behavior:

- If there are no associated activities just delete it from the database.
- If there are associated activities, just disable the account by adding a "disabled" boolean column, and make sure user gets no more emails.

Assigning to myself. @ariel-phet, please comment on the priority of this.

**Comment by ariel-phet** [comment]
@mattpen again low priority. We have the ability to delete an account if someone emails phethelp through the admin interface, so not a huge deal, but it will be good to give users this ability at some point.

**Comment by ariel-phet** [comment]
Changing this deferred priority, should be done at some point, but not until we are receiving regular requests for account deletion

**Comment by mattpen** [comment]
Discussed again at dev meeting on 5/5.  We will revisit the discussion of how to delete users and handle any detached activities or contributions after the new server goes live.

**Comment by ariel-phet** [comment]
@zepumph this issue is a bit more complicated than some of the small ones, but now that the new server architecture is working it would be good to handle.

**Comment by oliver-phet** [comment]
We had a few requests to delete accounts in the past week (perhaps triggered by email communications).

**Comment by zepumph** [comment]
I can get started on this but I would love to decide what the best thing to do would be. I think it makes sense to delete an account if they ask of that. So perhaps having one account we create—called PhET Activity Master, or something more real—would be a nice way to factor the activities out of the account to be deleted. 

As for the contact info that would be displayed on those activities. We could just put phethelp or no contact (or another option), I'm not sure which would be more useful. I don't think @oliver-phet would want queries about that but maybe I'm wrong. 

@ariel-phet @mattpen what do you think about this option. Do you think another would be easier/better/more efficient?

**Comment by mattpen** [comment]
@zepumph This approach sounds great.  phethelp@colorado.edu already has an account that we could use for this purpose.  We would probably want to bring it up to date first, it looks like there is no name or anything associated.

**Comment by kathy-phet** [comment]
Just a quick thought - we would want to keep the name associated with the activity, because attribution is important. Not sure how that would work!

**Comment by ariel-phet** [comment]
@zepumph a few things
- putting phethelp as the contact in these cases would be fine, queries are unlikely to be overwhelming
- As @kathy-phet mentioned we will want to retain the original author as the "author" and would only want to change the contact info
- In the activities database it seems we could decouple the contact info from the author (in most cases it would be the same but these two fields would not be strictly tied) or have some logic that says the contact info is the author contact info unless [condition]

**Comment by ariel-phet** [comment]
@zepumph moving to low priority at the moment, might be worth seeing if this work can be done in conjunction with updates to the registration pages, since you will be messing with the DB at that time.

**Comment by oliver-phet** [comment]
There is a 'Delete Account' button in the redesign on the Account Info page. https://github.com/phetsims/website/issues/547
